layers/LAYERS.org

```diff
@@ -67,64 +77,81 @@
+- [[file:+lang/latex/README.org][latex]]
+- [[file:+lang/markdown/README.org][markdown]]
+- [[file:+lang/plantuml/README.org][plantuml]]
+- [[file:+lang/restructuredtext/README.org][restructuredtext]]
+- [[file:+lang/yaml/README.org][yaml]]
+
+** General-purpose languages
+*** Imperative
+- [[file:+lang/asm/README.org][asm]]
+- [[file:+lang/forth/README.org][forth]]
+
+*** Multi-paradigm
 - [[file:+lang/c-c++/README.org][c-c++]]
-- [[file:+lang/clojure/README.org][clojure]]
-- [[file:+lang/common-lisp/README.org][common-lisp]]
-- [[file:+lang/coq/README.org][coq]]
 - [[file:+lang/csharp/README.org][csharp]]
-- [[file:+lang/csv/README.org][csv]]
 - [[file:+lang/d/README.org][d]]
 - [[file:+lang/elixir/README.org][elixir]]
-- [[file:+lang/elm/README.org][elm]]
-- [[file:+lang/emacs-lisp/README.org][emacs-lisp]]
 - [[file:+lang/erlang/README.org][erlang]]
-- [[file:+lang/ess/README.org][ess]]
-- [[file:+lang/faust/README.org][faust]]
-- [[file:+lang/forth/README.org][forth]]
 - [[file:+lang/fsharp/README.org][fsharp]]
 - [[file:+lang/go/README.org][go]]
-- [[file:+lang/gpu/README.org][gpu]]
-- [[file:+lang/graphviz/README.org][graphviz]]
-- [[file:+lang/haskell/README.org][haskell]]
-- [[file:+lang/html/README.org][html]]
-- [[file:+lang/idris/README.org][idris]]
-- [[file:+lang/ipython-notebook/README.org][ipython-notebook]]
 - [[file:+lang/java/README.org][java]]
-- [[file:+lang/javascript/README.org][javascript]]
 - [[file:+lang/jr/README.org][jr]]
-- [[file:+lang/latex/README.org][latex]]
 - [[file:+lang/lua/README.org][lua]]
-- [[file:+lang/markdown/README.org][markdown]]
 - [[file:+lang/nim/README.org][nim]]
 - [[file:+lang/ocaml/README.org][ocaml]]
-- [[file:+lang/octave/README.org][octave]]
 - [[file:+lang/perl5/README.org][perl5]]
 - [[file:+lang/perl6/README.org][perl6]]
 - [[file:+lang/php/README.org][php]]
-- [[file:+lang/plantuml/README.org][plantuml]]
-- [[file:+lang/protobuf/README.org][protobuf]]
-- [[file:+lang/purescript/README.org][purescript]]
 - [[file:+lang/python/README.org][python]]
 - [[file:+lang/racket/README.org][racket]]
-- [[file:+lang/restructuredtext/README.org][restructuredtext]]
 - [[file:+lang/ruby/README.org][ruby]]
 - [[file:+lang/rust/README.org][rust]]
 - [[file:+lang/scala/README.org][scala]]
-- [[file:+lang/scheme/README.org][scheme]]
-- [[file:+lang/shell-scripts/README.org][shell-scripts]]
 - [[file:+lang/sml/README.org][sml]]
-- [[file:+lang/sql/README.org][sql]]
 - [[file:+lang/swift/README.org][swift]]
+**** JavaScript
+- [[file:+lang/javascript/README.org][javascript]]
+- [[file:+lang/purescript/README.org][purescript]]
 - [[file:+lang/typescript/README.org][typescript]]
-- [[file:+lang/vimscript/README.org][vimscript]]
-- [[file:+lang/windows-scripts/README.org][windows-scripts]]
-- [[file:+lang/yaml/README.org][yaml]]
+
+**** Lisp dialects
+- [[file:+lang/clojure/README.org][clojure]]
+- [[file:+lang/common-lisp/README.org][common-lisp]]
+- Emacs Lisp: look [[Domail-specific languages]]
```

Comment:
Let's find a solution without cross-references. I guess that `Lisp dialects` is the most important category for `Emacs Lisp` so we can put it in this category only. Moreover we can say that any lisp is also a DSL as any lisp can be turned into a DSL easily.

Reply:
Done.